### ISSUE TITLE
🐛 Bug Report: Backstage utilities with "workspace" version specified in package.json cannot be installed by yarn

### ISSUE THREAD

**Issue by charlesreid1** [backstage/backstage#19746]
### 📜 Description

I am following the installation instructions on a Mac OS X machine. I have postgres installed and running in the background. When I try to install the postgres backend with this command,

```
# From your Backstage root directory
yarn add --cwd packages/backend pg
```

the build runs into a bunch of errors related to the version specified for backstage's utilities.

### 👍 Expected behavior

`yarn add --cwd packages/backend pg` should finish successfully, and should not complain that it can't find the package versions specified in package.json

### 👎 Actual Behavior with Screenshots

When I run the command, I see many, many, many complaints about the package version of various packages not being found. All of the packages have a version specified like `workspace:^` in the package.json file.

```
$ yarn add --cwd packages/backend pg
yarn add v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
Couldn't find any versions for "@backstage/types" that matches "workspace:^"
? Please choose a version of "@backstage/types" from this list: (Use arrow keys)
❯ 1.1.0
  1.0.2
  1.0.2-next.1
  1.0.2-next.0
  1.0.1
  1.0.1-next.0
  1.0.0
  0.1.3
  0.1.2
  0.1.1
  0.0.0-nightly-20230620022155
  0.0.0-nightly-20230619022652
  0.0.0-nightly-20230618023152
  0.0.0-nightly-20230617022331
...


Couldn't find any versions for "@backstage/core-components" that matches "workspace:^"
? Please choose a version of "@backstage/core-components" from this list: (Use arrow keys)
❯ 0.13.5-next.1
  0.13.5-next.0
  0.13.4
  0.13.4-next.0
  0.13.3
  0.13.3-next.2
  0.13.3-next.1
  0.13.3-next.0
  0.13.2
  0.13.2-next.3
  0.13.2-next.2
  0.13.2-next.1
  0.13.2-next.0
  0.13.1
  0.13.1-next.1
  0.13.1-next.0
  0.13.0
  0.13.0-next.3
  0.12.6-next.2
  0.12.6-next.1
  0.12.6-next.0

...

Couldn't find any versions for "@backstage/core-plugin-api" that matches "workspace:^"
? Please choose a version of "@backstage/core-plugin-api" from this list: (Use arrow keys)
❯ 1.6.0-next.1
  1.6.0-next.0
  1.5.3
  1.5.3-next.1
  1.5.3-next.0
  1.5.2
  1.5.2-next.0
  1.5.1
  1.5.1-next.1
  1.5.1-next.0
  1.5.0
  1.5.0-next.2
  1.4.1-next.1
  1.4.1-next.0
  1.4.0
  1.3.0
  1.3.0-next.1
  1.2.1-next.0
  1.2.0
  1.2.0-next.2
  1.1.1-next.1
  1.1.1-next.0
  1.1.0
  1.1.0-next.0
```

I can confirm I see the utilities that are supposed to be being installed, present in the `packages/` directory:

```
$ ls packages/
app
app-defaults
app-next
backend
backend-app-api
backend-common
backend-defaults
backend-dev-utils
backend-next
backend-openapi-utils
backend-plugin-api
backend-plugin-manager
backend-tasks
backend-test-utils
catalog-client
catalog-model
cli
cli-common
cli-node
codemods
config
config-loader
core-app-api

...
```

### 👟 Reproduction steps

On Mac 13.4.1, run the following commands:

```
brew install nvm
nvm install 18
nvm use 18
npm install --global yarn
yarn set version 1.22.19
npx @backstage/create-app@latest

cd my-backstage-app
yarn install
cd ../

brew install postgresql@12
brew services run postgresql@12

yarn add --cwd packages/backend pg
```

### 📃 Provide the context for the Bug.

Goal here is to successfully install the postgresql backend/plugin on a Mac.

### 🖥️ Your Environment

```
$ node --version
v18.17.1

$ yarn --version
1.22.19

$ yarn backstage-cli info
yarn run v1.22.19
error Command "backstage-cli" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by charlesreid1** [comment]
🤦  Just figured out the problem. When the instructions say "From your Backstage root directory", 

```
# From your Backstage root directory
yarn add --cwd packages/backend pg
```

they mean the directory of ***the backstage app*** that was just created. They do ***not*** mean the backstage repository root, which is where I was running the command.

Running the command in `my-backstage-app` directory instead of the backstage repo root worked like a charm.

**Comment by charlesreid1** [comment]
This issue will be addressed by #19748

**Comment by tudi2d** [comment]
thanks for adding the PR - linked it to close this issue on merge :)
